subprojects/core/src/main/java/org/gradle/caching/internal/controller/service/OpFiringBuildCacheServiceHandle.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.caching.internal.controller.service;
+
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.gradle.caching.BuildCacheKey;
+import org.gradle.caching.BuildCacheService;
+import org.gradle.caching.internal.controller.operations.DisableOperationDetails;
+import org.gradle.caching.internal.controller.operations.DisableOperationResult;
+import org.gradle.caching.internal.controller.operations.LoadOperationDetails;
+import org.gradle.caching.internal.controller.operations.LoadOperationHitResult;
+import org.gradle.caching.internal.controller.operations.LoadOperationMissResult;
+import org.gradle.caching.internal.controller.operations.StoreOperationDetails;
+import org.gradle.caching.internal.controller.operations.StoreOperationResult;
+import org.gradle.caching.internal.operations.BuildCacheDisableServiceBuildOperationType.Details.DisabledReason;
+import org.gradle.internal.operations.BuildOperationContext;
+import org.gradle.internal.operations.BuildOperationExecutor;
+import org.gradle.internal.operations.RunnableBuildOperation;
+import org.gradle.internal.progress.BuildOperationDescriptor;
+
+public class OpFiringBuildCacheServiceHandle extends BaseBuildCacheServiceHandle {
```

Comment:
Not in a way (that I can see) that makes things any better, other than dogmatic avoidance of inheritance.